packages/@ember/application/lib/application.js

```diff
@@ -387,6 +387,16 @@ const Application = Engine.extend({
     @return {ApplicationInstance} the application instance
   */
   buildInstance(options = {}) {
+    assert(
+      'You cannot build new instances of this application since it is being destroyed',
+      this.isDestroying
```

Comment:
We want to assert that we are not destroying when this method is called , should this be `!this.isDestroying`?

Reply:
🤦